packages/@uppy/companion/src/server/Uploader.js

```diff
@@ -219,12 +218,18 @@ class Uploader {
     if (chunk === null) {
       this.writeStream.on('finish', () => {
         this.streamsEnded = true
-        if (this.options.endpoint && protocol === PROTOCOLS.multipart) {
-          this.uploadMultipart()
-        }
-
-        if (protocol === PROTOCOLS.tus && !this.tus) {
-          return this.uploadTus()
+        switch (protocol) {
+          case PROTOCOLS.multipart:
+            if (this.options.endpoint) {
+              this.uploadMultipart()
+            }
+            break
+          case PROTOCOLS.s3Multipart:
+            this.uploadS3Multipart()
+            break
+          case PROTOCOLS.tus:
+            this.uploadTus()
```

Comment:
should we be extra cautious and keep the `if (!this.tus) {...}` conditional? Same for `if (!this. s3Upload) {...}` That way the class will not have to rely on trusting the downloader to not mess things up by calling `handleChunk(null)` twice.

Reply:
I didn't think it was necessary but I can't say I'm 100% sure. I'll add a warning log if that case happens too so that we can at least learn about it if it does occur